### ISSUE TITLE
Rework Custom Chatboxes

### ISSUE THREAD

**Issue by Jouto44** [AttorneyOnline/AO2-Client#865]
As of 2.10 RC1, Custom Chatboxes are still a mess to use, as shown in these images.
![image](https://user-images.githubusercontent.com/86204624/187049655-f8804ab0-2555-415f-8b7a-4c2a2f947de6.png)
**First image from Case Cafe, using 2.10 RC1**

![image](https://user-images.githubusercontent.com/86204624/187050458-82dd7e64-4f3d-4f16-a114-2fba2bd3cf0e.png)
**Second image from Vanilla, using 2.10 RC 1**

With Custom Chatboxes being broken still, it forces users to disable custom chatboxes due to how poorly implemented there are.

Optimally, it would be best to see these features reworked, so people can use Custom Chatboxes without issues.

**Proposed Solution to fixing Custom Chatboxes:**

Have the misc folder override the settings of  **ao2_chatbox**, **showname**, **showname_align**, **showname_extra_width**, and **message, chat_arrow** from a theme when a custom chatbox is used. Here's a sample of what it could look like:
![image](https://user-images.githubusercontent.com/86204624/187050630-a918ca86-5e85-4b0a-9f10-00453b1711ef.png)

If this solution is not feasible in anyway, then have the default settings have "Custom Chatboxes" set to off by default when a person get a fresh installation of AO until there is a solution to fix Custom Chatboxes.

**Comment by in1tiate** [comment]
Here's the issue. Custom chatboxes as a feature were introduced *before* widespread theming of the viewport size and thus chatbox size was a thing, and they're not built to accommodate it.

The solution is to include custom versions of these chatboxes within the theme itself that are tailored to the theme. This way, you can make sure the chatboxes your theme supports will always look correct on your theme.

It's not feasible to inherit default settings for custom chatboxes because the dimensions would be completely wrong unless the viewport was the same size (which it clearly isn't).

TL;DR: Either support custom chatboxes with your theme *explicitly*, or advise users of the theme to disable them.

**Comment by Salanto** [comment]
This would require a significant amount of changing how AO2s viewport works, cause frankly, I don't see the static nature that 2.10 clings to as being even remotely viable anymore without turning themes into discount Qt UI files.

**Comment by in1tiate** [comment]
> I'm begging you to give us _some_ way to fix this bull.

There is a way. Explicitly support these chatboxes with your theme. You can do this by copying them into `[YourTheme]/misc` and then adjusting the layout there to make sense with your theme. This will override the chatboxes in `base/misc` with the ones in your theme, which you have adjusted to play nice.
